operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/workflow/ResultCondition.java

```diff
@@ -0,0 +1,62 @@
+package io.javaoperatorsdk.operator.processing.dependent.workflow;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.api.reconciler.dependent.DependentResource;
+
+public interface ResultCondition<R, P extends HasMetadata, T> extends Condition<R, P> {
+  Result<T> detailedIsMet(DependentResource<R, P> dependentResource, P primary, Context<P> context);
+
+  @Override
+  default boolean isMet(DependentResource<R, P> dependentResource, P primary, Context<P> context) {
+    return detailedIsMet(dependentResource, primary, context).isSuccess();
+  }
+
+  interface Result<T> {
```

Comment:
Can't we have this as a builder rather? this feels a bit smelly to have it as an interface

Reply:
Not sure I'm following how a builder would help here.

Reply:
Can we see this working in an integration test, that might shed a light on it.

Reply:
See https://github.com/operator-framework/java-operator-sdk/pull/2426/commits/fd0e92c0de55c47d5df50658cf4e147ee5e6102d#diff-a2db869773190cffe048fc2b938d7808bf57160f274af63fde8e2b8cfc0443f8R188-R196